pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -0,0 +1,89 @@
+package admitters
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/api/admission/v1beta1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+type PodEvictionAdmitter struct {
+	ClusterConfig *virtconfig.ClusterConfig
+	VirtClient    kubecli.KubevirtClient
+}
+
+func (admitter *PodEvictionAdmitter) Admit(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	if !admitter.ClusterConfig.LiveMigrationEnabled() {
+		return allowed()
+	}
+
+	launcher, err := admitter.VirtClient.CoreV1().Pods(ar.Request.Namespace).Get(ar.Request.Name, metav1.GetOptions{})
+	if err != nil {
+		return allowed()
+	}
+
+	if value, exists := launcher.GetLabels()[virtv1.AppLabel]; !exists || value != "virt-launcher" {
+		return allowed()
+	}
+
+	domainName, exists := launcher.GetAnnotations()[virtv1.DomainAnnotation]
+	if !exists {
+		return allowed()
+	}
+
+	vmi, err := admitter.VirtClient.VirtualMachineInstance(ar.Request.Namespace).Get(domainName, &metav1.GetOptions{})
+	if err != nil {
+		return denied(fmt.Sprintf("kubevirt failed getting the vmi: %s", err.Error()))
+	}
+	if !vmi.IsEvictable() {
+		// we don't act on VMIs without an eviction strategy
+		return allowed()
+	} else if !vmi.IsMigratable() {
+		return denied(fmt.Sprintf(
```

Comment:
We propably shoud deny this but should we also suggest retry?

Reply:
It is not just for a retry.. it is to block the eviction process on the client's side until it completes. It is up to the client to decide what to do at this point. I agree that this "contract" with the client is not perfect here... it feels like we're missing some communication and error channels at this point, but that's what we have in our tool box right now. We might consider improving this in the future on `kubelet`.

Reply:
Maybe I misunderstand here something... but retry in this case would not make difference, right? So we should probably deny this but with a different response code so clients will not automaticly retry and the user will need to manual interfere.

Reply:
By "retry" I mean - this is our only way to block the client from proceeding and tell him that something is wrong and he should take a look at his VMI. He will then see the condition that his VMI is not migratable.

Reply:
Indeed. We definitely want to handle this better, but this is right now the only option which we have with k8s.